internal/engine/client.go

```diff
@@ -95,5 +96,14 @@ func waitBuildkit(ctx context.Context, host string) error {
 		}
 		time.Sleep(retryPeriod)
 	}
-	return errors.New("buildkit failed to respond")
+
+	// extract the most important error message from big output
+	pattern := "stderr="
```

Comment:
It's also possible for stdout to have relevant information here. I suspect it might just be best to trim the `\n` from the error but otherwise leave it alone. But maybe the full error is uglier than I am thinking, let me know if you disagree.

Reply:
I agree that stdout might have relevant information, it's just that it overloads the error message quite a bit:
<img width="1344" alt="image" src="https://user-images.githubusercontent.com/31691250/211048157-28fa1528-51e8-45a7-8d81-a984507c8e27.png">

As it's an error management log, and its aim is to help in case of issues, I understand your point of view. Rebasing the PR, and taking this into account 👌